src/tests/news/latest-mock.test.ts

```diff
@@ -0,0 +1,171 @@
+import { fetchFromNewsAPI } from '../../utils.js';
+import newsController from '../../modules/News/controller.js';
+
+jest.mock('../../utils.js');
+
+const mockFetchFromNewsAPI = fetchFromNewsAPI as jest.Mock;
+
+describe('news controller', () => {
+  it('should return articles with a 200 status code when fetchFromNewsAPI succeeds', async () => {
+    const mockArticles = {
+      articles: [{ title: 'News 1' }, { title: 'News 2' }],
+    };
+    mockFetchFromNewsAPI.mockResolvedValue(mockArticles);
+
+    const req = {
+      query: { q: 'test', limit: '10', sortBy: 'popularity' },
+    } as unknown as Request;
+    const res = {
+      status: jest.fn().mockReturnThis(),
+      send: jest.fn(),
+    } as unknown as Response;
+
+    await newsController.getLatestNews(req, res);
+
+    expect(mockFetchFromNewsAPI).toHaveBeenCalledWith('/everything', {
+      q: 'test',
+      limit: '10',
+      sortBy: 'popularity',
+    });
+    expect(res.status).toHaveBeenCalledWith(200);
+    expect(res.send).toHaveBeenCalledWith(mockArticles.articles);
+  });
+
+  it('should return a 500 status code with an error message when fetchFromNewsAPI throws an error', async () => {
+    const mockError = new Error('API Error');
+    mockFetchFromNewsAPI.mockRejectedValue(mockError);
+
+    const req = {
+      query: { q: 'test', limit: '10', sortBy: 'popularity' },
+    } as unknown as Request;
+    const res = {
+      status: jest.fn().mockReturnThis(),
+      send: jest.fn(),
+    } as unknown as Response;
+
+    await newsController.getLatestNews(req, res);
+
+    expect(mockFetchFromNewsAPI).toHaveBeenCalledWith('/everything', {
+      q: 'test',
+      limit: '10',
+      sortBy: 'popularity',
+    });
+    expect(res.status).toHaveBeenCalledWith(500);
+    expect(res.send).toHaveBeenCalledWith({
+      message: 'Error in getting the latest News',
+      error: mockError,
+    });
+  });
+});
+
+describe('/latest', () => {
+  const mockArticles = { articles: [{ title: 'News 1' }, { title: 'News 2' }] };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    mockFetchFromNewsAPI.mockResolvedValue(mockArticles);
+  });
+
+  describe('default parameters', () => {
+    it('should have default query limit parameters', async () => {
+      const defaultLimit = 5;
+      await request(app).get('/api/news/latest');
+
+      expect(mockFetchFromNewsAPI).toHaveBeenCalledTimes(1);
+      expect(mockFetchFromNewsAPI).toHaveBeenCalledWith(
+        '/everything',
+        expect.objectContaining({ limit: defaultLimit }),
+      );
+    });
+
+    it('should have default sortBy parameters', async () => {
+      const defaultSortBy = 'publishedAt';
+      await request(app).get('/api/news/latest');
+
+      expect(mockFetchFromNewsAPI).toHaveBeenCalledTimes(1);
+      expect(mockFetchFromNewsAPI).toHaveBeenCalledWith(
+        '/everything',
+        expect.objectContaining({ sortBy: defaultSortBy }),
+      );
+    });
+  });
+
+  describe('query parameters', () => {
```

Comment:
to differentiate this group from the next one maybe it should be called `'valid query parameters'`